warl_proposal/csr_template_draft1.yaml

```diff
@@ -0,0 +1,130 @@
+############################################ Template description ############################################
+# csr template for csrs with subfields
+<name>:                                   # name of the csr
+  description: <text>                     # textual description of the csr
+  address: <hex>                          # address of the CSR
+  priv_mode: <D/M/H/S/U>                  # privilege mode that owns the register
+  reset-val: <hex>                        # Reset value of the register. This an accumulation of the all reset values of the 
+                                          # sub-field 
+  rv32:                                   # this node and its subsequent fields can exist if [M/S/U]XL value can be 1
+    exists: <boolean>                     # indicates if the csr exists in rv32 mode or not. When False, all fields below 
+                                          # will be trimmed off in the checked yaml. False also indicates that 
+                                          # access-exception should be generated
+    subfields:                            # a quick summary of the list of all subfields of the csr including a list of 
+                                          # WPRI fields of the csr.
+      - <subfield_name1>
+      - <subfield_name2>
+      - wpri                              # contains a list of all WPRI fields within the csr. Can be empty as well
+        - [30-23]
+        - 6
+    <subfield_name1>:                     # name of the subfield
```

Comment:
I think we need bitmask for each field (optional, default is all ones). For example, for MTVEC.BASE field, user implementation may tie off the lower bits while the higher bits are still writable.

Reply:
Nevermind, I see it's covered by WARL specification